docs/content/_snippets/operator/profiling-flamegraphs.mdx

```diff
@@ -0,0 +1,37 @@
+## Profiling Flamegraphs
+
+Flame graphs provide a useful visualization of profiling data.
+The codebase heavily uses modular approach with message passing between various components.
+This makes analysis of flamegraphs difficult.
+The common tools (eg. `cargo flamegraph`) do not allow to distinguish between various components and asynchronous tasks.
+To alleviate this issue, we abuse code annotations with `tracing::span!` and `tracing::instrument`. 
+`telemetry-subscribers` crate implements `tracing::Subscriber` (and `tracing_subscriber::Layer`) capable of capturing time intervals which spans spend in "entered" and "exited" states per each thread/task.
+The "exited" state allows to capture time when a future (asynchronous task) is in idle/pending state and not doing any useful work.
+The profiling data captured by subscriber can be exposed via HTTP.
+Grafana Flamegraph panel is used to visualize profiling data captured by the subscriber.
+iota-node uses env variable `TRACE_FLAMEGRAPH` to enable the corresponding `tracing_subscriber::Layer` and exposes profiling data via `/flamegraph` admin server endpoint.
```

Comment:
Isn't the admin server only exposed to localhost, so if we run the node in a docker container, grafana wouldn't be able to access that endpoint?